### PR TITLE
Add systemd operation a `--user` operation mode

### PR DESCRIPTION
**Pull request by jprltsnz**
This adds a new switch to `pyinfra.operations.systemd.service` which would allow it to run on user mode `systemctl --user` which will allow to do user specific unit activation.

I am testing pyinfra to setup my personal machine on a "semi-reproducable" way and hit a limitation with my systemd user units.

### CODE REVIEW

File: pyinfra/operations/systemd.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def service(
     service,
     running=True, restarted=False, reloaded=False,
     command=None, enabled=None, daemon_reload=False,
-    state=None, host=None,
+    state=None, host=None, user_mode=False,
```
Comment (Fizzadar) [inline pyinfra/operations/systemd.py:28]: The `user_mode` argument here must live before `state` & `host`.

Sidenote: the documentation on this is _not_ explicit, I've added https://github.com/Fizzadar/pyinfra/issues/486 to track fixing that.

**Review by Fizzadar — CHANGES_REQUESTED** [review]
Hi @jprltsnz, thank you for the PR! One little thing to fix then this is good to go :)

File: pyinfra/operations/systemd.py (hunk 1 above)
Comment (jprltsnz) [reply pyinfra/operations/systemd.py]: Done!

**Comment by jprltsnz** [comment]
@Fizzadar should be good to go now.

**Review by Fizzadar — APPROVED** [review]
Thank you @jprltsnz!

**Pull request merged**